### PR TITLE
Improvements to the README doc

### PR DESCRIPTION
**Pull request by mvanchaa**
Following updates are made:
1. AzureAuth currently doesn't support token caching in Linux environments. MSAL under the hood doesn't support token caching yet. Updating the README accordingly. 
2. On Ubuntu, the web auth flow requires a non-headless environment. Updated it.
3. Remove the reference to an archived MSAL extension repo.
4. Add IWA auth flow to the `platform X auth flow support` table.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ The CLI is designed for authenticating and returning an access token for public
 | ------------------------------------------ | ----------------------- | ------------- | ---------------- | ------------- | ------------------------------- |
 | Windows                                    | ✅                      | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
 | OSX (MacOS)                                | ⚠️ via Web Browser      | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
-| Ubuntu (linux) <br/>‼️Releases coming soon | ⚠️ via Edge             | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
+| Ubuntu (linux) <br/>‼️Releases coming soon | ⚠️ via Edge             | ✅            | ✅               | ❎         | ⚠️ `--domain` account filtering |
```
Comment (mijpeterson) [inline README.md:20]: Is there a red X that we can use? I almost didn't notice that it's an X.

**Review by mijpeterson — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mvanchaa) [reply README.md:20]: Totally agreed! We couldn't find a red X :(

Comment (mijpeterson) [reply README.md:20]: Would this one do okay? ❌

**Review by keyuxuan — DISMISSED** [review]

**Review by mijpeterson — APPROVED** [review]

**Review by Haard30 — DISMISSED** [review]

**Review by evandigby — APPROVED** [review]

**Review by Haard30 — APPROVED** [review]

**Review by mijpeterson — DISMISSED** [review]

**Review by mijpeterson — APPROVED** [review]

**Review by emilyyang-ms — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -7,17 +7,17 @@
...
-| Ubuntu (linux) <br/>‼️Releases coming soon | ⚠️ via Edge             | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
+| Operating System                           | Integrated Windows Auth | Auth Broker Integration | Web Auth Flow            | Device Code Flow | Token Caching | Multi-Account Support           |
+| ------------------------------------------ | ----------------------- | ----------------------- | ------------------------ | ---------------- | ------------- | ------------------------------- |
+| Windows                                    | ✅                      | ✅                      | ✅                      | ✅              | ✅          | ⚠️ `--domain` account filtering |
+| OSX (MacOS)                                | ❌                      | ⚠️ via Web Browser      | ✅                      | ✅              | ✅          | ⚠️ `--domain` account filtering |
+| Ubuntu (linux) <br/>‼️Releases coming soon  | ❌                      | ⚠️ via Edge             | ⚠️ via non headless env | ✅              | ❌          | ⚠️ `--domain` account filtering |
```
Comment (reillysiemens) [inline README.md:20]:
```suggestion
| Ubuntu (linux) <br/>‼️Releases coming soon  | ❌                      | ⚠️ via Edge             | ⚠️ via non headless env | ✅              | ⚠️ in GUI environments. See [msal/#3033](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3033)          | ⚠️ `--domain` account filtering |
```

File: README.md [hunk 3]
```diff
@@ -7,17 +7,17 @@
...
-| OSX (MacOS)                                | ⚠️ via Web Browser      | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
-| Ubuntu (linux) <br/>‼️Releases coming soon | ⚠️ via Edge             | ✅            | ✅               | ✅            | ⚠️ `--domain` account filtering |
+| Operating System                           | Integrated Windows Auth | Auth Broker Integration | Web Auth Flow            | Device Code Flow | Token Caching | Multi-Account Support           |
+| ------------------------------------------ | ----------------------- | ----------------------- | ------------------------ | ---------------- | ------------- | ------------------------------- |
+| Windows                                    | ✅                      | ✅                      | ✅                      | ✅              | ✅          | ⚠️ `--domain` account filtering |
+| OSX (MacOS)                                | ❌                      | ⚠️ via Web Browser      | ✅                      | ✅              | ✅          | ⚠️ `--domain` account filtering |
```
Comment (reillysiemens) [inline README.md:19]:
```suggestion
| OSX (MacOS)                                | N/A                      | ⚠️ via Web Browser      | ✅                      | ✅              | ✅          | ⚠️ `--domain` account filtering |
```

**Review by reillysiemens — CHANGES_REQUESTED** [review]

**Review by reillysiemens — APPROVED** [review]

**Review by Haard30 — APPROVED** [review]

**Review by keyuxuan — APPROVED** [review]

**Review by mijpeterson — APPROVED** [review]

**Pull request merged**